eox_nelp/signals/tests/test_tasks.py

```diff
@@ -76,7 +85,48 @@ def test_dispatch_futurex_progress(self, post_futurex_progress_mock, generate_pr
             course_id=course_id,
             user_has_passing_grade=True,
         )
-        post_futurex_progress_mock.assert_called_with(data)
+        post_futurex_progress_mock.assert_called_with(progress_enrollment_data)
+
+    @patch("eox_nelp.signals.tasks._generate_progress_enrollment_data")
+    @patch("eox_nelp.signals.tasks._post_futurex_progress")
+    def test_not_call_dispatch_logic_setting_not_configured(
+        self, post_futurex_progress_mock, generate_progress_enrollment_data_mock
+    ):
+        """Test when course_grade_changed_progress_publisher is not called
+        due no setting configured.
+
+        Expected behavior:
+            - dispatch_futurex_progress is not called due settings.
+        """
+        user, _ = User.objects.get_or_create(username="vader")
+        course_id = "course-v1:test+Cx105+2022_T4"
+
+        dispatch_futurex_progress(course_id, user.id, is_complete=True)
+
+        generate_progress_enrollment_data_mock.assert_not_called()
+        post_futurex_progress_mock.assert_not_called()
+
+    @patch("eox_nelp.signals.tasks.settings")
+    @patch("eox_nelp.signals.tasks._generate_progress_enrollment_data")
+    @patch("eox_nelp.signals.tasks._post_futurex_progress")
+    @data(*FALSY_ACTIVATION_VALUES)
+    def test_not_call_dispatch_logic_setting_setting_falsy(
+        self, falsy_value, post_futurex_progress_mock, generate_progress_enrollment_data_mock, settings_mock
+    ):
+        """Test when course_grade_changed_progress_publisher is not called
+        due no setting configured.
+
+        Expected behavior:
+            - dispatch_futurex_progress is not called due settings.
```

Comment:
the method is called but the logic is not executed

Reply:
[5b2d1cf](https://github.com/eduNEXT/eox-nelp/pull/41/commits/5b2d1cf6d3b48ace51ae229da337b6bd82aa659b)